### ISSUE TITLE
Resolution to #177 : Workaround or release (Intermittent PHP Invalid lang ID)

### ISSUE THREAD

**Issue by ricjoh** [PDConSec/vsc-print#185]
Issue #177 you said you have a fix for the next release. Do you know when that will be? Is there a workaround to have a valid syntax Lang identifier going into the print so it doesn't require auto? Why is it only some tabs/files?

**Comment by PeterWone** [comment]
> Do you know when that will be? 

No. I'm working on it, but I have a day job and two large properties to look after.

> Is there a workaround to have a valid syntax Lang identifier going into the print so it doesn't require auto? 

Not currently, but that's quite a good idea: a way to explicitly map the vscode language identifier to a highlightjs language identifier when they differ. The biggest problem is it requires the user to understand the nature of the problem and figure out what both sides are using to define the mapping. 

If you want to make a contribution, you could compile a list of mappings. If you did that, I would certainly incorporate this into Print, but it won't make the next release.

> Why is it only some tabs/files?
That's not specific enough for me to speculate.

**Comment by PeterWone** [comment]
I had to back out the fix. Highlightjs v11 has a problem with its Typescript type definitions that borked webpack and I had to go back to 10.7.3

However, after further consideration of #177 and your observations in this issue, I have some ideas that may resolve it.

**Comment by PeterWone** [comment]
Could you give me some PHP sample files? One of each different file extension in use, ideally ones that manifest the problem.

**Comment by PeterWone** [comment]
For mapping 
https://github.com/highlightjs/highlight.js/blob/main/SUPPORTED_LANGUAGES.md
https://code.visualstudio.com/docs/languages/identifiers
